File: 1.0/src/cz/mg/vulkan/PFNvkGetMemoryFdPropertiesKHR.java
package cz.mg.vulkan;

public class PFNvkGetMemoryFdPropertiesKHR extends VkFunctionPointer {
    public PFNvkGetMemoryFdPropertiesKHR() {
    }

    protected PFNvkGetMemoryFdPropertiesKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetMemoryFdPropertiesKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetMemoryFdPropertiesKHR(long value) {
        setValue(value);
    }

    public PFNvkGetMemoryFdPropertiesKHR(VkInstance instance) {
        super(instance, new VkString("vkGetMemoryFdPropertiesKHR"));
    }

    public int call(VkDevice device, int handleType, int fd, VkMemoryFdPropertiesKHR pMemoryFdProperties){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), handleType, fd, pMemoryFdProperties != null ? pMemoryFdProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, int handleType, int fd, long pMemoryFdProperties);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetMemoryFdPropertiesKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetMemoryFdPropertiesKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong handleType, jlong fd, jlong pMemoryFdProperties, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetMemoryFdPropertiesKHR f = (PFN_vkGetMemoryFdPropertiesKHR)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkExternalMemoryHandleTypeFlagBits*)jniLongToPointer(handleType)),
        *((int*)jniLongToPointer(fd)),
        ((VkMemoryFdPropertiesKHR*)jniLongToPointer(pMemoryFdProperties))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetMemoryFdPropertiesKHR_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jint handleType, jint fd, jlong pMemoryFdProperties){
    (void)env;
    (void)jc;
    PFN_vkGetMemoryFdPropertiesKHR f = (PFN_vkGetMemoryFdPropertiesKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        (VkExternalMemoryHandleTypeFlagBits)handleType,
        (int)fd,
        ((VkMemoryFdPropertiesKHR*)jniLongToPointer(pMemoryFdProperties))
    );
}

